packages/blitz/scripts/postinstall.js

```diff
@@ -0,0 +1,346 @@
+  }
+  return null
+}
+
+async function main() {
+  if (process.env.INIT_CWD) {
+    process.chdir(process.env.INIT_CWD) // necessary, because npm chooses __dirname as process.cwd()
+    // in the postinstall hook
+  }
+  await ensureEmptyDotBlitz()
+
+  const localPath = getLocalPackagePath()
+  // Only execute if !localpath
+  const installedGlobally = localPath ? undefined : await isInstalledGlobally()
+
+  // this is needed, so that the Generate command does not fail in postinstall
+
+  process.env.BLITZ_GENERATE_IN_POSTINSTALL = "true"
+
+  // this is needed, so we can find the correct schemas in yarn workspace projects
+  const root = findPackageRoot(localPath)
+
+  process.env.BLITZ_GENERATE_IN_POSTINSTALL = root ? root : "true"
+
+  debug({
+    localPath,
+    installedGlobally,
+    init_cwd: process.env.INIT_CWD,
+    BLITZ_GENERATE_IN_POSTINSTALL: process.env.BLITZ_GENERATE_IN_POSTINSTALL,
+  })
+  try {
+    if (localPath) {
+      await run("node", [
+        localPath,
+        "codegen",
+        "--postinstall",
+        doubleQuote(getPostInstallTrigger()),
+      ])
+      return
+    }
+    if (installedGlobally) {
+      await run("blitz", ["codegen", "--postinstall", doubleQuote(getPostInstallTrigger())])
+      return
+    }
+  } catch (e) {
+    // if exit code = 1 do not print
+    if (e && e !== 1) {
+      console.error(e)
+    }
+    debug(e)
+  }
+
+  if (!localPath && !installedGlobally) {
+    console.error(
+      `In order to use "@blitz/client", please install Blitz CLI. You can install it with "npm add -D blitz".`,
+    )
+  }
+}
+
+function getLocalPackagePath() {
+  try {
+    const packagePath = require.resolve("blitz/package.json")
+    if (packagePath) {
+      return require.resolve("blitz")
+    }
+  } catch (e) {
+    //
+  }
+
+  try {
+    const packagePath = require.resolve("@blitz/cli/package.json")
+    if (packagePath) {
+      return require.resolve("@blitz/cli")
+    }
+  } catch (e) {}
+
+  return null
+}
+
+async function isInstalledGlobally() {
+  try {
+    await exec("blitz -v")
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+if (!process.env.BLITZ_SKIP_POSTINSTALL_GENERATE) {
+  main()
+    .catch((e) => {
+      console.error(e)
+      process.exit(0)
+    })
+    .finally(() => {
+      debug(`postinstall trigger: ${getPostInstallTrigger()}`)
+    })
+}
+
+function run(cmd, params, cwd = process.cwd()) {
+  const child = childProcess.spawn(cmd, params, {
+    stdio: ["pipe", "inherit", "inherit"],
+    cwd,
+  })
+
+  return new Promise((resolve, reject) => {
+    child.on("close", () => {
+      resolve()
+    })
+    child.on("exit", (code) => {
+      if (code === 0) {
+        resolve()
+      } else {
+        reject(code)
+      }
+    })
+    child.on("error", () => {
+      reject()
+    })
+  })
+}
+
+async function ensureEmptyDotBlitz() {
+  try {
+    const dotBlitzDir = path.join(process.cwd(), "node_modules/.blitz")
```

Comment:
We need to take the same approach as prisma here so it works correctly with monorepos.

```suggestion
    const dotBlitzDir = path.join(__dirname, "../../.blitz")
```

Reply:
`__dirname` is not compatible with Blitz's preconstruct setup, it'd generate into the wrong folder. What's the best way to detect that?

```ts
const dotBlitzDir =
  process.env.IS_IN_BLITZ_REPO // Do we already have a detection mechanism for smth like this?
    ? path.join(process.cwd(), "node_modules/.blitz")
    : path.join(__dirname, "../../.blitz")
```

Reply:
The postinstall script on latest canary takes care of this (used for nextjs fork). So check how I do it there.

Reply:
Resolved in https://github.com/blitz-js/blitz/pull/2042/commits/10d7afc124bf6fdd47cbb1a6695046e00e1a07dc